inc/generation.class.php

```diff
@@ -39,7 +39,7 @@
 class PluginGeninventorynumberGeneration {
 
    static function autoName($config, CommonDBTM $item) {
-      $template = $config['template'];
+      $template = htmlentities(html_entity_decode($config['template']));
```

Comment:
```suggestion
      $template = htmlentities(Toolbox::clean_cross_side_scripting_deep($config['template']));
```

Reply:
the generation seems incorrect with this way

![image](https://user-images.githubusercontent.com/7335054/121650076-7f54c100-ca99-11eb-9697-85df134098aa.png)

I should have

X000001

Reply:
My mistake, I suggest to use "clean", but it should be "unclean".
The unclean method is made to revert the clean process and has been made to handle old and new clean logic.

```suggestion
      $template = htmlentities(Toolbox::unclean_cross_side_scripting_deep($config['template']));
```

Reply:
ok for me